harness/determined/launch/deepspeed.py

```diff
@@ -129,36 +130,43 @@ def create_sshd_cmd() -> List[str]:
     ]
 
 
-def create_deepspeed_env_file() -> None:
-    """Create an env var export file to pass Determined vars to the deepspeed launcher.
+def filter_env_vars(env: Mapping[str, str]) -> Mapping[str, str]:
+    """
+    Calculate all non-dangerous environment variables that we can pass to training workers.
 
     By default, the deepspeed launcher only keeps env vars that start with one of the following
     ["NCCL", "PYTHON", "MV2", "UCX"].
 
-    There are certain variables that we need to be set that we can pass to deepspeed using
-    a custom env vars file.
+    We modify this behavior to include all environment variables except for ones we know to be
+    problematic.  This is the same strategy taken by horovodrun.
     """
-    INCLUDE = [
-        "PATH",
-        "LD_LIBRARY_PATH",
-        "USE_DEEPSPEED",
-        "DET_CHIEF_IP",
-        "DET_MANUAL_INIT_DISTRIBUTED",
-        "DET_DEEPSPEED_HOSTFILE_PATH",
-        "DET_MASTER_CERT_FILE",
-        "DET_MASTER_CERT_NAME",
+
+    EXCLUDE_REGEX = [
+        "BASH_FUNC_.*",
+        "OLDPWD",
+        "HOSTNAME",
+        ".*CUDA_VISIBLE_DEVICES",
+        "SLURM_PROCID",
+        "DET_SLOT_IDS",
+        "DET_AGENT_ID",
     ]
+
+    excludes = [re.compile(x) for x in EXCLUDE_REGEX]
+
```

Comment:
nit: Can we also log a warning message about any env vars that do get excluded in this process? Think it would help avoid headaches and confusion if any expected env vars go missing.

Reply:
yeah that's a great idea

Reply:
Though after thinking about it, I left the log level at debug because this script would _always_ print some things, and those log lines would be totally useless >99% of the time.

Reply:
yeah, whatever level is fine, just as long as there's some reasonable way the message might be passed along